File: code/string_algorithms/src/lipogram_checker/lipogram_checker.cs
using System;
/*
 * Part of cosmos by OpenGenus Foundation
 * */
namespace lipogram_checker
{
    class Program
    {
        /// <summary>
        /// Method to check if a string is lipogram or not
        /// </summary>
        /// <param name="str">String To Check</param>
        /// <returns>Integer Value</returns>
        static int lipogram(string str)
        {
            int count = 0;
            string alphabets = "abcdefghijklmnopqrstuvwxyz";
            str = str.ToLower();
            for (int i = 0; i < alphabets.Length; i++)
            {
                int index = str.IndexOf(alphabets[i]);

                if (index < 0 || index > str.Length)
                    count++;
            }
            return count;
        }
        static void Main(string[] args)
        {
            string to_chk = "The quick brown fox jumped over the lazy dog";

            int val = lipogram(to_chk);

            if (val == 0)
                Console.WriteLine("Pangram");
            else if (val >= 2)
                Console.WriteLine("Not a pangram but might a lipogram");
            else
                Console.WriteLine("Pangrammatic Lipogram");

            Console.ReadKey();
        }
    }
}

File: code/string_algorithms/src/lipogram_checker/lipogram_checker.cpp
// Part of Cosmos by OpenGenus Foundation
#include <iostream>
#include <string>

const std::string alphabets = "abcdefghijklmnopqrstuvwxyz";

void
panLipogramChecker(std::string phrase)
{
    using std::cout;
    using std::endl;

    for (unsigned int i = 0; i < phrase.length(); i++)
        phrase[i] = tolower(phrase[i]);
    int counter = 0;
    for (unsigned int i = 0; i < 26; i++)
    {
        auto pos = phrase.find(alphabets[i]);

        if (pos > phrase.length())
            counter += 1;
    }

    if (counter == 0)
        cout << "Pangram" << endl;
    else if (counter >= 2)
        cout << "Not a pangram but might be a lipogram" << endl;
    else
        cout << "Pangrammatic Lipogram" << endl;
}

int
main()
{
    std::string str = "The quick brown fox jumped over the lazy dog";
    panLipogramChecker(str);

    str = "The quick brown fox jumps over the lazy dog";
    panLipogramChecker(str);

    str = "The quick brown fox jum over the lazy dog";
    panLipogramChecker(str);
}
